tools/k4aviewer/k4aimugraph.cpp

```diff
@@ -121,7 +121,8 @@ void K4AImuGraph::PlotGraph(const char *name,
     std::stringstream nameBuilder;
     nameBuilder << "##" << name;
 
-    const float currentData = graphData[static_cast<size_t>(graphFrontIdx)].v[offset];
+    const float currentData =
+        graphData[(static_cast<size_t>(graphFrontIdx) + graphData.size() - 1) % graphData.size()].v[offset];
```

Comment:
@williampMSFT I just assumed this was a ring buffer, and it seems to work, but I couldn't really find any docs on the `ImGui::PlotLines` function. Hopefully you can tell me if my assumptions were correct.

Reply:
Yep, it's a ring buffer.

---
In reply to: [303670547](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/505#discussion_r303670547) [](ancestors = 303670547)